server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -289,7 +293,8 @@ public Query rangeQuery(
                 boolean includeLower,
                 boolean includeUpper,
                 boolean hasDocValues,
-                SearchExecutionContext context
+                SearchExecutionContext context,
+                boolean forceDocValues
```

Comment:
nit: I can see why the additional argument is called forceDocValues here, because we explicitly want to query only doc_values, but here too maybe isIndexed would also express that if the field is indexed we query that or doc_values, while if it's not indexed we can only query doc_values.

Reply:
ok, fixed in 4b273fd94c1